lib/iex/lib/iex/helpers.ex

```diff
@@ -277,30 +278,50 @@ defmodule IEx.Helpers do
   end
 
   @doc """
-  Prints the history of expressions evaluated during the session along with
-  their results.
+  Retrieves the last expression's value from the history
   """
   def v do
-    inspect_opts = IEx.inspect_opts
-    IEx.History.each(history, &print_history_entry(&1, inspect_opts))
-  end
-
-  defp print_history_entry({counter, cache, result}, inspect_opts) do
-    IO.write IEx.color(:eval_info, "#{counter}: #{cache}#=> ")
-    IO.puts  IEx.color(:eval_result, "#{inspect result, inspect_opts}\n")
+    v(-1)
   end
 
   @doc """
-  Retrieves the nth expression's value from the history.
+  Retrieves expression values from the history.
+
+  Use `:all` to retrieve all values from the history.
+
+  Use a range to retrieve values from the history.
+
+  Use an integer to retrieve the nth value from the history.
 
   Use negative values to lookup expression values relative to the current one.
   For instance, v(-1) returns the result of the last evaluated expression.
   """
-  def v(n) do
+  def v(:all), do: v(1..-1)
```

Comment:
Ah, you misunderstood my interpretation of `v(:all)` but that is ok because your implementation actually makes much more sense. :D However, it just made me realize it is not super useful so I would rather remove it altogether. Can we get rid of `v(:all)`,  `v(range)` and `vs()`? Let's just remove code while we can it, I don't think anybody will miss those features. :) Thank you!